### ISSUE TITLE
Make LayoutKit extension safe

### ISSUE THREAD

**Issue by FabianTerhorst** [linkedin/LayoutKit#43]
This line is not compatible with today extension. https://github.com/linkedin/LayoutKit/blob/master/LayoutKit/LayoutArrangement.swift#L105

Could this get checked like http://stackoverflow.com/questions/29246656/ios-sharedapplication-is-unavailable-in-today-extension to add support for extensions as well?

**Comment by nicksnyder** [comment]
It would be nice for LayoutKit to be extension safe.

If we just skip this code then it seems like right-to-left languages wouldn't be handled properly in the today extension. Need to find out if there is an extension safe way to know if the UI should be right-to-left.

**Comment by FabianTerhorst** [comment]
What's about: ```
if #available(iOS 9.0, *) {
  if UIView.userInterfaceLayoutDirection(
    for: myView.semanticContentAttribute) == .rightToLeft {


      // The view is shown in right-to-left mode right now.
  }
} else { /* Use the previous technique */ }
```

**Comment by nicksnyder** [comment]
@FabianTerhorst Can you share what error you are seeing and when. I just tried adding a Today extension to LayoutKitSampleApp and (surprisingly) everything seemed to work fine.

**Comment by FabianTerhorst** [comment]
@nicksnyder im getting  ``` 'sharedApplication' is unavailable: not available on iOS (App Extension) ``` as a build error. At the bottom of the page: https://developer.apple.com/library/ios/documentation/General/Conceptual/ExtensibilityPG/ExtensionOverview.html its also documented that this can´t work.

**Comment by nicksnyder** [comment]
I am able to reproduce this in a separate app.

This is a compile time error so the `else` case can't use `Application.sharedApplication` either.

This means we have to choose between supporting right-to-left languages on Today Extensions and supporting it on iOS 8.

I think what we can do is add a build flag so consumers can make this decision for themselves.

```
#if LAYOUTKIT_EXTENSION_SAFE
    if #available(iOS 9.0, *) {
        // TODO
    } else {
        // can't support right to left languages here :(
    }
#else
    if Application.sharedApplication().userInterfaceLayoutDirection != direction {
        flipSubviewsHorizontally(view)
    }
#endif
```

Will need to fixup the sample app and add a Today extension so we can test this.

**Comment by nicksnyder** [comment]
For the record, the final solution won't require a build flag
